File: mapbiomas/get-sentinel-masks.py

import ee

ee.Initialize()

assetPathRow = "users/joaovsiqueira1/sentinel-2-acquisition-plans"
assetGrids = "projects/mapbiomas-workspace/AUXILIAR/cartas"

assetOutput = "projects/mapbiomas-workspace/AUXILIAR/sentinel-2-mask"

grids = ee.FeatureCollection(assetGrids)
pathRow = ee.FeatureCollection(assetPathRow)

gridNames = grids.aggregate_array('grid_name').getInfo()

for gridName in gridNames:

    grid = grids.filter(ee.Filter.eq('grid_name', gridName))\
        .geometry()\
        .buffer(300)\
        .bounds()

    pathRowsSelected = pathRow.filter(ee.Filter.bounds(grid))

    pathRowsId = pathRowsSelected.aggregate_array('OrbitAbsol').getInfo()

    for pathRowId in pathRowsId:

        pathRowFeature = pathRowsSelected.filter(
            ee.Filter.eq('OrbitAbsol', pathRowId))

        raster = ee.Image().byte().paint(
            featureCollection=pathRowFeature,
            color=1
        )

        assetid = gridName + '-' + str(pathRowId)

        print(assetid)

        task = ee.batch.Export.image.toAsset(
            image=raster.byte().set('grid_name', gridName).set('orbit', pathRowId),
            description=assetid,
            assetId=assetOutput + '/' + assetid,
            region=grid.coordinates().getInfo(),
            scale=10,
            maxPixels=int(1e13)
        )

        task.start()

File: mapbiomas/get-sentinel-masks.js

var assetPathRow = "users/joaovsiqueira1/sentinel-2-acquisition-plans";
var assetOrbit = "projects/mapbiomas-workspace/AUXILIAR/sentinel-2-orbit";

var geometry = ee.Geometry.Polygon(
    [
        [
            [-82.91763689923195, 13.496542967765722],
            [-82.91763689923195, -57.46855036794329],
            [-31.765293149231972, -57.46855036794329],
            [-31.765293149231972, 13.496542967765722]
        ]
    ], null, false);

var pathRow = ee.FeatureCollection(assetPathRow)
    .filter(ee.Filter.bounds(geometry));

var orbits = pathRow.aggregate_histogram('OrbitRelat').keys().getInfo();

print(orbits);

orbits.forEach(
    function (orbit) {

        var raster = ee.Image().paint(pathRow.filter(ee.Filter.eq('OrbitRelat', orbit)), 1);

        Map.addLayer(raster.multiply(ee.Number.parse(orbit)).randomVisualizer());

        Export.image.toAsset({
            image: raster.set('orbit', orbit),
            description: 'orbit-' + orbit,
            assetId: assetOrbit + '/' + 'orbit-' + orbit,
            pyramidingPolicy: { '.default': 'MODE' },
            region: geometry,
            scale: 10,
            maxPixels: 1e13
        }
        );
    }
);